From: Hans Rosenfeld <hans.rosenfeld@amd.com>
Subject: x86, amd: Extract compute unit information for AMD CPUs
References: fate#309723
Patch-mainline: v2.6.37-rc1

Signed-off-by: Thomas Renninger <trenn@suse.de>

Get compute unit information from CPUID Fn8000_001E_EBX.
(See AMD CPUID Specification - publication # 25481, revision 2.34,
September 2010.)

Note that each core on a compute unit still has a core_id of its own.

Signed-off-by: Andreas Herrmann <andreas.herrmann3@amd.com>
LKML-Reference: <20100930123857.GE20545@loge.amd.com>
Signed-off-by: H. Peter Anvin <hpa@linux.intel.com>

Signed-off-by: Hans Rosenfeld <hans.rosenfeld@amd.com>
Automatically created from "patches.arch/0022-x86-amd-Extract-compute-unit-information-for-AMD-CPU.patch" by xen-port-patches.py

--- sle11sp2-2011-06-28.orig/arch/x86/include/mach-xen/asm/processor.h	2011-06-28 13:58:24.000000000 +0200
+++ sle11sp2-2011-06-28/arch/x86/include/mach-xen/asm/processor.h	2011-06-28 14:02:38.000000000 +0200
@@ -119,6 +119,8 @@ struct cpuinfo_x86 {
 	u16			phys_proc_id;
 	/* Core id: */
 	u16			cpu_core_id;
+	/* Compute unit id */
+	u8			compute_unit_id;
 #endif
 #ifdef CONFIG_SMP
 	/* Index into per_cpu list: */
